tests/unit/test_bruteforce.cpp

```diff
@@ -1751,15 +1751,15 @@ TEST_F(BruteForceTest, rangeQueryCosine) {
     for (size_t i = 0; i < dim; i++) {
         query[i] = 1.0f;
     }
-    auto verify_res = [&](size_t id, float score, size_t index) {
+    auto verify_res = [&](size_t id, double score, size_t index) {
         ASSERT_EQ(id, index + 1);
-        float first_coordinate = float(n - index) / n;
+        double first_coordinate = double(n - index) / n;
         // By cosine definition: 1 - ((A \dot B) / (norm(A)*norm(B))), where A is the query vector
         // and B is the current result vector.
-        float expected_score =
+        double expected_score =
             1.0f -
-            ((first_coordinate + (float)dim - 1.0f) /
-             (sqrtf((float)dim) * sqrtf((float)(dim - 1) + first_coordinate * first_coordinate)));
+            ((first_coordinate + (double)dim - 1.0f) /
```

Comment:
```suggestion
            ((first_coordinate + (double)dim - 1.0) /
```

also in other places, remove the `f` (which stands for float) where it isn't necessary..